### PR TITLE
Fix some small bugs of filter methods (part 1)

### PR DESCRIPTION
**Pull request by keroro520**
I have added some test cases, which helped me identify some bugs: https://github.com/nervosnetwork/godwoken-tests/pull/143

This PR brings some small bugfixes for them.

### CODE REVIEW

File: packages/api-server/src/methods/modules/eth.ts [hunk 1]
```diff
@@ -989,7 +1002,7 @@ export class Eth {
       }
 
       const fromBlockNumber: U64 = await this.blockParameterToBlockNumber(
-        filter.fromBlock || "earliest"
+        filter.fromBlock || "latest"
```
Comment (RetricSu) [inline packages/api-server/src/methods/modules/eth.ts:1005]: not sure if this is in conflict with https://github.com/nervosnetwork/godwoken-web3/pull/300/files/131542bd5cc272279d27760e258fb5fa5de6fc9a#diff-cce75f92de0f01f77430b0195c78d8a547ca11f2b8f818202008dce92a69ebef ?

cc @classicalliu

**Review by RetricSu — COMMENTED** [review]
other parts LGTM

File: packages/api-server/src/methods/modules/eth.ts (hunk 1 above)
Comment (classicalliu) [reply packages/api-server/src/methods/modules/eth.ts:1005]: Why change to `latest` block here ?

Comment (keroro520) [reply packages/api-server/src/methods/modules/eth.ts:1005]: In order to make it compatible, I've changed it to `latest`. Here are Geth and Alchemy declaring that `fromBlock` and `toBlock` should be `latest`:
  - https://github.com/ethereum/go-ethereum/blob/10dc5dce0871bf8c24bac41b04e47c3b9ad2b93e/eth/filters/api.go#L280
  - https://docs.alchemy.com/alchemy/apis/ethereum/eth-newfilter#parameters

Comment (classicalliu) [reply packages/api-server/src/methods/modules/eth.ts:1005]: Have you test this issue https://github.com/nervosnetwork/godwoken-web3/issues/273 ?

It's `fromBlock` should be latest block when `newFilter` , not the latest block when `getFilter`

`earliest` will always lower than id cursor, so the true from is the id cursor, if this be `latest`, `from` will always equals to `to` if no `fromBlock` & `toBlock` provided. And `id` to current latest block will be lost, so it changed from `latest` to `earliest`

Comment (keroro520) [reply packages/api-server/src/methods/modules/eth.ts:1005]: @classicalliu Thanks for your correction. Yes, you are right. Unlike Geth and other projects that use subscriptions, we implement the filter by querying the database.

I will change it back to "earliest".
Would you mind if I put your response into the code comment?

Comment (keroro520) [reply packages/api-server/src/methods/modules/eth.ts]: [8e22fd0](https://github.com/nervosnetwork/godwoken-web3/pull/427/commits/8e22fd099896cc0d1da0848971399a9f29037d81) reverted changes. PTAL. @classicalliu @RetricSu

Comment (classicalliu) [reply packages/api-server/src/methods/modules/eth.ts]: > @classicalliu Thanks for your correction. Yes, you are right. Unlike Geth and other projects that use subscriptions, we implement the filter by querying the database. They don't have to "get filter".
> 
> I will change it back to "earliest". Would you mind if I put your response into the code comment?

Of course not

**Review by classicalliu — APPROVED** [review]

File: packages/api-server/src/methods/modules/eth.ts [hunk 2]
```diff
@@ -1031,7 +1047,10 @@ export class Eth {
     // remember to update the last poll cache
     // logsData[0] is now the highest log id(meaning it is the newest cache log id)
```
Comment (RetricSu) [inline packages/api-server/src/methods/modules/eth.ts:1048]:
```suggestion
    // logsData[logsData.length -1] is now the highest log id(meaning it is the newest cache log id)
```

**Review by RetricSu — APPROVED** [review]

**Pull request merged**